plugin renaming preparation, SVG data-URLs support and others

AmjadHD opened the pull request:
Image Preview v0.2.0
===========================

## Breaking Changes

- the plugin was renamed to "Image Preview" (previously "Hover Image Preview")
- the settings file was renamed accordingly

## New Features

- added support for SVG data URLs (#29)

## Fixes

- [data-]URL images on long lines where not previewed (#40)

## Notes

- the "max_chars" settings was removed, it was more trouble than it worth
- the default value for "image_folder_name" setting is now "previewed_images" (previously "Hovered Images")

AmjadHD committed:
Rename the plugin

- rename "Hover Image Preview.sublime-settings" ==>\
   "Image Preview.sublime-settings"
- rename `main.hover_preview_callback` ==> `main.image_preview_callback`
- changed the default value for the "image_folder_name` setting to\
 "previewed_images" (previously "hovered images")

AmjadHD committed:
Don't trim the line

AmjadHD committed:
Only call `get_image_size()` once

- rename `get_dimensions()` ==> `get_data()`
- reuse `real_width` and `real_height` to calculate the adjusted\
 `width` and `height`

AmjadHD committed:
Add v0.2.0 message

AmjadHD committed:
Make the description more visible

AmjadHD committed:
More consistent settings handling

- add a new file settings.py to utils
- add a Settings class that contains all the settings to settings.py\
that handles updating them with it's `update()` method thus separating\
settings from other variables
- rename `image_preview_callback()` ==> `on_change()`
- Make `Settings` attributes and variables that change when settings\
change lowercase since these are not considered constants

AmjadHD committed:
Add more mypy type hints

I found out that theres no need for a "typing" dependency

AmjadHD committed:
Refactor `get_data()`

AmjadHD committed:
Silence more mypy warnings

AmjadHD committed:
Don't ignore .gitignore

AmjadHD committed:
Add a .sublime-project file

AmjadHD committed:
Set the maximum line length to 120

AmjadHD committed:
Remove View import

AmjadHD committed:
Ignore "Previewed Images" folder

AmjadHD committed:
Fix bug in `convert()`

* `Settings` has no attribute `all_formats`

AmjadHD committed:
Filter out ST supported formats from those to convert

* don't replace the extension with "png" if it's supported by ST
* add a note in the settings

AmjadHD committed:
Add some commented out debugging statements

AmjadHD committed:
Remove unused variables

AmjadHD committed:
Better names for some variables

AmjadHD committed:
Add support for SVG data URLs

AmjadHD committed:
Merge remote-tracking branch 'upstream/master'

AmjadHD committed:
Make "Previewed Images" lowercase with underscores

AmjadHD committed:
Fix #41

AmjadHD committed:
Some cleanups

AmjadHD committed:
Add pyls setting for the Project

AmjadHD committed:
Remove an unused import

AmjadHD commented:
@alvesjtiago, can you review this.

alvesjtiago commented on main.py:
```diff
@@ -25,60 +29,55 @@
+    ST_FORMATS = {"png", "jpg", "jpeg", "bmp", "gif"}
+    unique_formats_to_convert = set(Settings.formats_to_convert)
+    all_formats = list(ST_FORMATS.union(unique_formats_to_convert))
+    # filter ou ST supported formats
```
Missing a `t` on the word out?

AmjadHD replied:
Done!

alvesjtiago commented on main.py:
```diff
@@ -25,60 +29,55 @@
+    all_formats = list(ST_FORMATS.union(unique_formats_to_convert))
+    # filter ou ST supported formats
+    formats_to_convert = tuple('.' + ext for ext in unique_formats_to_convert - ST_FORMATS)
+    # print("[Image Preview]", all_formats, formats_to_convert, unique_formats_to_convert)
```
Consider removing commented log.

AmjadHD replied:
Done!

alvesjtiago reviewed (commented):
Changes look great, thank you so much @AmjadHD 🙏
Do you ming fixing those two small comments and I'll prepare the migration to the new name on package control.

AmjadHD committed:
Remove pycodestyle entry from pyls settings

AmjadHD committed:
Fix a typo and remove a comment

AmjadHD committed:
Mention fix of #41

AmjadHD committed:
Change the name of the image folder

__previewed_images__ is less likely to be used

AmjadHD committed:
Rename the plugin to "ImagePreview"

according to the [naming guidelines]\
(https://packagecontrol.io/docs/submitting_a_package#Step_2)
a package name should be in PascalCase or snail_case.

AmjadHD committed:
Remove another occurrence of "Preview Image"

AmjadHD committed:
Add support for python 3.8 (and ST4)

alvesjtiago commented:
@AmjadHD I've just submitted a PR to Package Control (https://github.com/wbond/package_control_channel/pull/7799). As soon as it's merged I'll merge this PR as well. Thank you

AmjadHD commented:
Hhh, finally.

AmjadHD commented:
@alvesjtiago, i noticed you didn't make the changes in https://github.com/alvesjtiago/hover-preview/issues/39#issuecomment-549359514

alvesjtiago commented:
@AmjadHD changing it right now. I will not be changing the repo URL for now. I'll do it in the next version not to disrupt the whole process.

alvesjtiago commented:
Done - https://github.com/wbond/package_control_channel/pull/7799/commits/b8723a451c025307863410f4a540102ce31f75ba

AmjadHD commented:
Hi @alvesjtiago, https://github.com/wbond/package_control_channel/pull/7799 has been merged.
